### PR TITLE
enh: dataframe and series `.item()`

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Closes: Some related comment lost in the ether
 
## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

## If you have comments or can explain your changes, please do so below.

TIL I've learned... we don't have `series.head()`

### CODE REVIEW

File: narwhals/_pandas_like/dataframe.py [hunk 1]
```diff
@@ -314,3 +314,21 @@ def null_count(self: Self) -> PandasDataFrame:
             self._dataframe.isnull().sum(axis=0).to_frame().transpose(),
             implementation=self._implementation,
         )
+
+    def item(self: Self, row: int | None = None, column: int | str | None = None) -> Any:
```
Comment (FBruzzesi) [inline narwhals/_pandas_like/dataframe.py:318]: From [polars itself](https://github.com/pola-rs/polars/blob/py-0.20.31/py-polars/polars/dataframe/frame.py#L1218-L1267)

File: narwhals/_pandas_like/utils.py [hunk 2]
```diff
@@ -270,12 +270,17 @@ def reuse_series_namespace_implementation(
     )
 
 
-def item(s: Any) -> Any:
+def item(s: Any, index: int | None = None) -> Any:
```
Comment (FBruzzesi) [inline narwhals/_pandas_like/utils.py:273]: So.. I didn't want to move it directly to the pandas_like series because I noticed it is used in grouped object as well, which I am not confident in touching at almost midnight.

Implementation also follows polars one

File: tests/test_series.py [hunk 3]
```diff
@@ -504,3 +505,21 @@ def test_cast_string() -> None:
...
+df_pandas = pd.DataFrame({"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]})
+
+
+@pytest.mark.parametrize("df_raw", [df_pandas, df_polars])
+@pytest.mark.parametrize(("index", "expected"), [(0, 1), (1, 3)])
+def test_item(df_raw: Any, index: int, expected: int) -> None:
+    s = nw.Series(df_raw["a"])
+    result = s.item(index)
+    assert result == expected
+    assert nw.Series(df_raw["a"].head(1)).item() == 1
```
Comment (FBruzzesi) [inline tests/test_series.py:519]: Here's where I learned we don't support `series.head()` and used this workaround. It should be quick to implement anyway on a separate PR

File: narwhals/_pandas_like/utils.py (hunk 2 above)
Comment (FBruzzesi) [reply narwhals/_pandas_like/utils.py]: Edit: Moved it to series directly, nothing breaks in tests 👌

**Review by MarcoGorelli — APPROVED** [review]
awesome stuff, well done @FBruzzesi ! feel free to merge
